### DISCUSSION TITLE
Cake.Build doesn't generate description for nuget and non-sdk projecs

### DISCUSSION THREAD

**Discussion by DmitryLukyanov** [cake-build/cake#4293] [Q&A]
have quite simple nuget package creating logic with cake.build:

            var nugetSettings = new NuGetPackSettings
            {
                OutputDirectory = "%some_path%",
                Verbosity = NuGetVerbosity.Detailed,
                Description = "oops, description",
                Version = "2.0.0"
            };

            context.NuGetPack("%project_path%", nugetSettings);

It generates nuget package and set expected version, but the description value is ignored. This is what I still see in ClassLibrary1.2.0.0.nupkg:
![image](https://github.com/cake-build/cake/assets/44870738/5da461eb-aa04-497e-a705-89a0fb03d2a6)

Target framework:

       <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>

Description value is not appeared in the package metadata even if I manually add Description property to the .csproj file like:

       <Description>asd</Description>

This question is continuation of the discussion from [here ](https://stackoverflow.com/questions/78077589/cake-build-doesnt-generate-description-for-nuget-and-non-sdk-projecs?noredirect=1#comment137649844_78077589). It looks like Description even in theory can't be filled in this case? If so, why Description field does exist?

**Comment by nils-a** [comment]
The short answer is that `NuGetPack` (like nuget.exe) has two very distinct usages: One operating on a project file and one operating on a `.nuspec` file and not all settings that are available in the `NuGetPackSettings` are used for both usages.

I created https://github.com/cake-build/cake/issues/4295 to track that issue.

**Reply by DmitryLukyanov** [reply]
How to know which of these ways I use when I run `NuGetPack`?

**Reply by nils-a** [reply]
Well, if you call `NuGetPack` with the first parameter pointing to a `.csproj`/`.vbproj`/`.fsproj` you are using the "project" way. For every other file we assume the file is a `.nuspec` file.
